src/sapling/saplingscriptpubkeyman.cpp

```diff
@@ -671,6 +671,9 @@ isminetype SaplingScriptPubKeyMan::IsMine(const CWalletTx& wtx, const SaplingOut
 
 CAmount SaplingScriptPubKeyMan::GetCredit(const CWalletTx& tx, const isminefilter& filter, const bool fUnspent) const
 {
+    if (!tx.IsShieldedTx() || tx.sapData->vShieldedOutput.empty()) {
```

Comment:
`IsShieldedTx` is asking for `vShieldedOutput.empty` inside. Would be good if we add a new method `hasShieldedOutputs`.
(Same for the `GetDebit`, an `hasShieldedSpends`)